components/application-broker/internal/knative/subscription.go

```diff
@@ -0,0 +1,102 @@
+package knative
+
+import (
+	"log"
+	"regexp"
+
+	apicorev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	knapisv1alpha1 "knative.dev/pkg/apis/v1alpha1"
+
+	messagingv1alpha1 "knative.dev/eventing/pkg/apis/messaging/v1alpha1"
+)
+
+const (
+	// maxPrefixLength for limiting the max length of the name prefix
+	maxPrefixLength = 10
+
+	// generatedNameSeparator for adding a separator after the generated name prefix
+	generatedNameSeparator = "-"
+)
+
+type SubscriptionBuilder struct {
+	subscription *messagingv1alpha1.Subscription
+}
+
+func Subscription(prefix, namespace string) *SubscriptionBuilder {
+	// format the name prefix
+	prefix = formatPrefix(prefix, generatedNameSeparator, maxPrefixLength)
+
+	// construct the Knative Subscription object
+	subscription := &messagingv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: prefix,
+			Namespace:    namespace,
+		},
+	}
+
+	// init a new Subscription builder
+	return &SubscriptionBuilder{subscription: subscription}
+}
+
+func FromSubscription(subscription *messagingv1alpha1.Subscription) *SubscriptionBuilder {
+	// init a new Subscription builder
+	return &SubscriptionBuilder{subscription: subscription}
+}
+
+func (b *SubscriptionBuilder) Spec(channel *messagingv1alpha1.Channel, subscriberURI string) *SubscriptionBuilder {
+	url, err := apis.ParseURL(subscriberURI)
+	if err != nil {
+		panic("todo: nils, return error in build()")
+	}
+	b.subscription.Spec = messagingv1alpha1.SubscriptionSpec{
+		Channel: apicorev1.ObjectReference{
+			Name:       channel.Name,
+			Kind:       channel.Kind,
+			APIVersion: channel.APIVersion,
+		},
+		Subscriber: &knapisv1alpha1.Destination{
+			URI: url,
+		},
+	}
+	return b
+}
+
+func (b *SubscriptionBuilder) Labels(labels map[string]string) *SubscriptionBuilder {
+	if len(labels) == 0 {
+		return b
+	}
+	if len(b.subscription.Labels) == 0 {
+		b.subscription.Labels = labels
+		return b
+	}
+	for k, v := range labels {
+		b.subscription.Labels[k] = v
+	}
+	return b
+}
+
+func (b *SubscriptionBuilder) Build() *messagingv1alpha1.Subscription {
```

Comment:
Why does this method is called Build?

Reply:
Follow up PR: https://github.com/kyma-project/kyma/pull/6728/files